lib/stages/ReadGain.cpp

```diff
@@ -0,0 +1,142 @@
+#include "ReadGain.hpp"
+
+#include "buffer.h"
+#include "bufferContainer.hpp"
+#include "chimeMetadata.h"
+#include "configUpdater.hpp"
+#include "errors.h"
+#include <utils/visUtil.hpp>
+#include <functional>
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::configUpdater;
+using kotekan::Stage;
+
+using kotekan::connectionInstance;
+using kotekan::HTTP_RESPONSE;
+using kotekan::restServer;
+
+REGISTER_KOTEKAN_STAGE(ReadGain);
+
+ReadGain::ReadGain(Config& config, const std::string& unique_name,
+                           bufferContainer& buffer_container) :
+  Stage(config, unique_name, buffer_container, std::bind(&ReadGain::main_thread, this)) {
+  //Apply config.
+  _num_elements = config.get<uint32_t>(unique_name, "num_elements");
+  scaling = config.get_default<float>(unique_name, "frb_scaling", 1.0);
+  vector<float> dg = {0.0, 0.0}; // re,im
+  default_gains = config.get_default<std::vector<float>>(unique_name, "frb_missing_gains", dg);
+
+  metadata_buf = get_buffer("network_buf");
+  register_consumer(metadata_buf, unique_name.c_str());
```

Comment:
Hmm, this is actually a little tricky with the current framework.  You only need one of these buffers to get the metadata, but once you register as a consumer it will expect you to read every frame from this buffer, otherwise it will stay "full".   I think this actually requires a bit of a framework tweek, let me get back to you on this.